js/web/lib/wasm/wasm-core-impl.ts

```diff
@@ -68,16 +69,24 @@ const activeSessions = new Map<number, SessionMetadata>();
 
 /**
  * allocate the memory and memcpy the model bytes, preparing for creating an instance of InferenceSession.
- * @returns a 2-elements tuple - the pointer and size of the allocated buffer
+ * @returns a 3-elements tuple - the pointer, size of the allocated buffer, and optional weights.pb FS node
  */
-export const createSessionAllocate = (model: Uint8Array): [number, number] => {
+export const createSessionAllocate = (model: Uint8Array, weights?: ArrayBuffer): [number, number, FSNode?] => {
   const wasm = getInstance();
   const modelDataOffset = wasm._malloc(model.byteLength);
   if (modelDataOffset === 0) {
     throw new Error(`Can't create a session. failed to allocate a buffer of size ${model.byteLength}.`);
   }
   wasm.HEAPU8.set(model, modelDataOffset);
-  return [modelDataOffset, model.byteLength];
+
+  let weightsFile: FSNode|undefined;
+  if (weights) {
+    weightsFile = wasm.FS.create('/home/web_user/weights.pb');
+    weightsFile.contents = new Uint8Array(weights);
```

Comment:
This will be changed later once I make a PR to emscripten to have proxy objects for files >4gb